Author: Volker Krause <vkrause@kde.org>
Date:   Sun Oct 25 18:22:58 2020 +0100

    Compile with recent Qt 5.15
    
    More QPrinter methods got deprecated there. DevicePixel no longer exists
    as a unit, visually Point looks pretty much the same though.

diff --git src/printing/compact/compactstyle.cpp src/printing/compact/compactstyle.cpp
index d950a963..71e74c0f 100644
--- src/printing/compact/compactstyle.cpp
+++ src/printing/compact/compactstyle.cpp
@@ -168,7 +168,7 @@ void CompactStyle::print(const KContacts::Addressee::List &contacts, PrintProgre
 
     // print
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff --git src/printing/grantlee/grantleeprintstyle.cpp src/printing/grantlee/grantleeprintstyle.cpp
index d7afe8ef..49b5e494 100644
--- src/printing/grantlee/grantleeprintstyle.cpp
+++ src/printing/grantlee/grantleeprintstyle.cpp
@@ -38,7 +38,7 @@ GrantleePrintStyle::~GrantleePrintStyle()
 void GrantleePrintStyle::print(const KContacts::Addressee::List &contacts, PrintProgress *progress)
 {
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff --git src/printing/mike/mikesstyle.cpp src/printing/mike/mikesstyle.cpp
index 9f698c53..81a4bd6f 100644
--- src/printing/mike/mikesstyle.cpp
+++ src/printing/mike/mikesstyle.cpp
@@ -107,7 +107,7 @@ MikesStyle::~MikesStyle()
 void MikesStyle::print(const KContacts::Addressee::List &contacts, PrintProgress *progress)
 {
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff --git src/printing/ringbinder/ringbinderstyle.cpp src/printing/ringbinder/ringbinderstyle.cpp
index 60a39c86..5f600d6a 100644
--- src/printing/ringbinder/ringbinderstyle.cpp
+++ src/printing/ringbinder/ringbinderstyle.cpp
@@ -159,7 +159,7 @@ void RingBinderPrintStyle::print(const KContacts::Addressee::List &contacts, Pri
     config.sync();
 
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(50, 20, 0, 50, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(50, 20, 0, 50), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
